src/groups/mqb/mqbblp/mqbblp_queueengineutil.h

```diff
@@ -540,10 +541,11 @@ struct QueueEngineUtil_AppsDeliveryContext {
                                bmqp::Protocol::SubQueueInfosArray>
         Consumers;
 
-    Consumers              d_consumers;
-    bool                   d_doRepeat;
-    mqbi::StorageIterator* d_currentMessage;
-    mqbi::Queue*           d_queue_p;
+    Consumers                d_consumers;
+    bool                     d_doRepeat;
+    mqbi::StorageIterator*   d_currentMessage;
+    mqbi::Queue*             d_queue_p;
+    bsl::vector<bsl::string> d_active_appIds;
```

Comment:
This is performance-critical path.  Since appId outlives `QueueEngineUtil_AppsDeliveryContext`, we can keep references instead of copies.